core/class/jMQTT.class.php

```diff
@@ -1529,7 +1529,41 @@ public function getBroker() {
         }
         return $this->_broker;
     }
-    
+
+    /**
+     * Check if a certificate file is used by the Broker
+     * If TLS is disabled, gracefully replaces the certificate by the default configuration value.
+     * @param string $certname Certificate file name
+     * @return boolean true if certificat is used
+     */
+    public function isCertUsed($certname) {
+        if ($certname == $this->getConf(self::CONF_KEY_MQTT_TLS_CA)) {
+            if (boolval($this->getConf(self::CONF_KEY_MQTT_TLS))) {
+                return true;
+            } else {
+                $this->setConfiguration(self::CONF_KEY_MQTT_TLS_CA, $this->getDefaultConfiguration(self::CONF_KEY_MQTT_TLS_CA));
+                $this->save();
+            }
+        }
+        if ($certname == $this->getConf(self::CONF_KEY_MQTT_TLS_CLI_CERT)) {
+            if (boolval($this->getConf(self::CONF_KEY_MQTT_TLS))) {
```

Comment:
And so this condition can be moved out around 3 if

Reply:
Yep, in fact this only a big boolean operation now ;)